### ISSUE TITLE
NOT_IMPLEMENTED : Could not find an implementation for Trilu(14) node with name 'Trilu_2

### ISSUE THREAD

**Issue by Slyne** [microsoft/onnxruntime#9494]
**Describe the bug**
NOT_IMPLEMENTED : Could not find an implementation for Trilu(14) node with name 'Trilu_2

**System information**
- OS Platform and Distribution (Linux Ubuntu 16.04):
- ONNX Runtime installed from (binary):
- ONNX Runtime version:  1.9.0
- Python version: 3.8
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**To Reproduce**
```python
import torch
import onnxruntime

class trilModel(torch.nn.Module):
    def forward(self, a):
        size = a.size(0)
        ret = torch.ones(size, size, dtype=torch.bool)
        return torch.tril(ret)

model = trilModel()
x = torch.ones(10)
torch.onnx.export(model, x, 'tril.onnx',opset_version=14, do_constant_folding=True)
providers=["CUDAExecutionProvider"]
ort_session = onnxruntime.InferenceSession('tril.onnx', providers=providers)
ort_inputs = {ort_session.get_inputs()[0].name: x.numpy()}
ort_outs = ort_session.run(None, ort_inputs)
print(ort_outs)
```

**Comment by flowpoint** [comment]
Hi,
i hit the same issue with a big gpt-model.
the attention masks are also implemented in this way.
i am not working on a production model, but i'd think that boolean attention masks could be much more efficent.
it would also ease conversion of any models implemented this way.
greetings

**Comment by robertknight** [comment]
I am trying to export the T5 model from [torchtext](https://github.com/pytorch/text) and have run into this issue. Below is a Python script that generates the ONNX model.

I am using PyTorch v2.1.0, torchtext v0.16.0.

```py
import torch
from torchtext.models import (
    T5_SMALL_GENERATION,
    T5_BASE_GENERATION,
    T5_LARGE_GENERATION,
)

PADDING_TOKEN = 0
EOS_TOKEN = 1


def export_model(model, encoder_tokens, decoder_tokens):
    torch.onnx.export(
        model,
        {"encoder_tokens": encoder_tokens, "decoder_tokens": decoder_tokens},
        "t5-torchtext.onnx",
        input_names=["encoder_tokens", "decoder_tokens"],
        output_names=["logits"],
        dynamic_axes={
            "encoder_tokens": {0: "batch", 1: "enc_seq_len"},
            "decoder_tokens": {0: "batch", 1: "dec_seq_len"},
        },
    )


def test_model(model, encoder_tokens, decoder_tokens):
    while decoder_tokens[0, -1] != EOS_TOKEN:
        outputs = model(encoder_tokens=encoder_tokens, decoder_tokens=decoder_tokens)
        new_toks = outputs["decoder_output"].argmax(-1)
        decoder_tokens = torch.cat([decoder_tokens, new_toks[:, -1:]], dim=1)
    return decoder_tokens


t5_base = T5_SMALL_GENERATION
transform = t5_base.transform()
input_str = "translate to french: studies have shown that owning a dog is good for you"

encoder_tokens = transform(input_str)
encoder_tokens = encoder_tokens.unsqueeze(0)

model = t5_base.get_model()
decoder_tokens = torch.tensor([[PADDING_TOKEN]])
export_model(model, encoder_tokens, decoder_tokens)

# output = test_model(model, encoder_tokens, decoder_tokens)
# print("Input:", input_str)
# print("Input tokens:", encoder_tokens[0].tolist())
# print("Output tokens:", output[0])
# decoded = transform.decode(output[0].tolist())
# print("Output: ", decoded)
```
